WordPress/openverse-frontend
# Remove `mob` breakpoint and simplify VFilterButton

## Fixes


Actually there are no issues for this. The files I renamed to TS were already type checked.

The removal of the `mob` breakpoint happened in a private discussion between @dhruvkb and myself about why it existed that was prompted by me trying to refine the types for the media query composable. It turns out the `mob` breakpoint was redundant with the `xs` so we removed it.

This also prompted some clean up with breakpoint modifiers in the templates which then led me to removing a ton of JavaScript from the VFilterButton and replacing it with breakpoint-based conditional CSS :tada: This makes the SSR rendering & hydration of the filter button much more pleasant.

Also because the button no longer only uses JavaScript to determine how to present itself, I moved the tests to visual regression tests. It still requires some programatic stuff for some things, so I added a utility `gotoWithArgs` in the storybook-specific playwright utils that makes it easier to write tests against a story with specific arguments passed to it.

This also prompted me to clean up the filter button story, it had some confusing stuff in it that didn't appear to be doing anything (probably just left over from copy/pasting another story I'd imagine).

## Description


* Renames several files to TS and moves the types from JSDoc to TypeScript
* Removes the `mob` breakpoint in favor of just using xs as the lowest (which was the lowest in Tailwind anyway).
* Fixes usages of the `xs` breakpoint modifier in the templates. These were basically noop modifications because `xs` is the lowest breakpoint and is always met (unless a device goes under 340 but there are no known devices used today that do that we would need to target specifically anyway)
* Removes a ton of JavaScript-based conditional style rendering from the VFilterButton to improve the code and the rendering in SSR matching the client without a jump after the page hydration. As much of this kind of JavaScript as we can remove the better. Some of it isn't possible to remove (like the button variant one, unfortunately, though I keep trying to think of a way to make that work) but any of it that can be removed should be.

## Testing Instructions

Checkout the branch and compare it to staging, visually. It should be nearly identical with the only differences showing in the filter button in the SSR render before hydration (there should be marginally less jumpiness now).

Be sure to test applying filters and scrolling or not scrolling, etc at various page widths. I've written snapshot tests for these cases, if I missed one let me know and I will add it.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
